src/components/chart/ha-network-graph.ts

```diff
@@ -0,0 +1,273 @@
+  };
+  ignoreForceLayout?: boolean;
+}
+
+export interface NetworkData {
+  nodes: NetworkNode[];
+  links: NetworkLink[];
+  categories?: { name: string }[];
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+let GraphChart: any;
+
+@customElement("ha-network-graph")
+export class HaNetworkGraph extends LitElement {
+  public chart?: EChartsType;
+
+  @property({ attribute: false }) public data?: NetworkData;
+
+  @property({ attribute: false }) public tooltipFormatter?: (
+    params: TopLevelFormatterParams
+  ) => string;
+
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _reducedMotion = false;
+
+  @state() private _physicsEnabled = true;
+
+  private _listeners: (() => void)[] = [];
+
+  private _nodePositions: Record<string, { x: number; y: number }> = {};
+
+  @query("ha-chart-base") private _baseChart?: HaChartBase;
+
+  constructor() {
+    super();
+    if (!GraphChart) {
+      import("echarts/lib/chart/graph/install").then((module) => {
+        GraphChart = module;
+        this.requestUpdate();
+      });
+    }
+  }
+
+  public async connectedCallback() {
+    super.connectedCallback();
+    this._listeners.push(
+      listenMediaQuery("(prefers-reduced-motion)", (matches) => {
+        if (this._reducedMotion !== matches) {
+          this._reducedMotion = matches;
+        }
+      })
+    );
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    while (this._listeners.length) {
+      this._listeners.pop()!();
+    }
+  }
+
+  shouldUpdate(changedProperties: PropertyValues) {
+    // don't update if only hass changes
+    return changedProperties.size > 1 || !changedProperties.has("hass");
+  }
+
+  protected render() {
+    if (!GraphChart) {
+      return nothing;
+    }
+    return html`<ha-chart-base
+      .hass=${this.hass}
+      .data=${this._getSeries()}
+      .options=${this._createOptions()}
+      height="100%"
+      .extraComponents=${[GraphChart]}
+    >
+      <slot name="button" slot="button"></slot>
+      <ha-icon-button
+        slot="button"
+        class="refresh-button ${this._physicsEnabled ? "active" : "inactive"}"
+        .path=${mdiGoogleCirclesGroup}
+        @click=${this._togglePhysics}
+        title=${this.hass.localize(
+          "ui.panel.config.common.graph.toggle_physics"
+        )}
+      ></ha-icon-button>
+    </ha-chart-base>`;
+  }
+
+  private _createOptions(): ECOption {
+    return {
+      tooltip: {
+        trigger: "item",
+        confine: true,
+        formatter: this.tooltipFormatter,
+      },
+      legend: {
+        show: !!this.data?.categories?.length,
+        data: this.data?.categories,
+        top: 8,
+      },
+      dataZoom: {
+        type: "inside",
+        filterMode: "none",
+      },
+    };
+  }
+
+  private _getSeries() {
+    if (!this.data) {
+      return [];
+    }
+
+    const containerWidth = this.clientWidth;
+    const containerHeight = this.clientHeight;
```

Comment:
Should we add a resize controller then?

Reply:
I had added it but then I tried it without and it still worked fine so I removed it. I think it's because we only use this for the initial layout and after that echarts takes over and handles the resize.